pkg/kubelet/dockertools/docker_manager_test.go

```diff
@@ -144,17 +154,26 @@ func createTestDockerManager(fakeHTTPClient *fakeHTTP, fakeDocker *FakeDockerCli
 	return dockerManager, fakeDocker
 }
 
+func createTestDockerManagerWithFakeImageManager(fakeHTTPClient *fakeHTTP, fakeDocker *FakeDockerClient) (*DockerManager, *FakeDockerClient) {
+	dm, fd := createTestDockerManager(fakeHTTPClient, fakeDocker)
+	dm.imagePuller = newFakeImageManager()
+	return dm, fd
+}
+
+func newTestDockerManagerWithRealImageManager() (*DockerManager, *FakeDockerClient) {
```

Comment:
Hm, seems like we need a builder for fake docker client and fake docker manager.
`TestDockerManagerBuilder().HTTPClient(xx).ImageManager(xx)....Build()`

We can do this cleanup in the future. :)